src/collectiveMemory/remember.ts

```diff
@@ -9,13 +10,55 @@ export const HDRConfig = z.object({
 
 export type HDRConfig = z.infer<typeof HDRConfig>;
 
+export async function fetchStateActionPairs(
+  state: ObjectiveState,
+  hdrConfig: HDRConfig,
+  limit: number = 2
+) {
+  const { apiKey, endpoint } = hdrConfig;
+
+  const response = await fetch(`${endpoint}/memories/remember`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${apiKey}`,
+    },
+    body: JSON.stringify({
+      state: ObjectiveState.parse(state),
+      limit,
+    }),
+  });
+
+  if (!response.ok) {
+    throw new Error(
+      `HDR API request failed with status ${response.status} to url ${response.url}`
+    );
+  }
+  const data = await response.json();
+
+  return data.map((m: any) =>
+    Memory.parse({
+      actionStep: m.action,
+      objectiveState: m.state,
+    })
+  ) as Memory[];
+}
+
 // TODO: fill out
```

Comment:
?

Reply:
The TS compiler doesn't register this unless I explicitly set the type. I can fix in the morning tho

Reply:
the `//TODO: fill out` because you did fill it out